src/version/version.go

```diff
@@ -0,0 +1,214 @@
+package version
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/NeowayLabs/semantic-release/src/style"
+	"github.com/NeowayLabs/semantic-release/src/time"
+)
+
+const (
+	major                    = "MAJOR"
+	minor                    = "MINOR"
+	patch                    = "PATCH"
+	versionPattern           = "%d.%d.%d"
+	versionPyVariable        = "__version__"
+	commitMsgLimitCharacters = 150
+	authorChangelogFormat    = "@%s"
+
+	ErrMsgConvertToInt                   = "could not convert %v to int"
+	ErrMsgInvalidUpgradeChangeType       = "%s is an invalid upgrade change type"
+	ErrMsgSplitVersion                   = "error while spliting version into MAJOR.MINOR.PATCH due to: %s"
+	ErrMsgGetUpgradeType                 = "error while getting upgrade type due to: %s"
+	ErrMsgGetChangeType                  = "change type not found"
+	ErrMsgGetCommitChangeTypeFromMessage = "error while finding commit change type within commit message due to: %s"
+)
+
+var (
+	commitChangeTypes             = []string{"build", "ci", "docs", "fix", "feat", "perf", "refactor", "style", "test", "breaking change", "breaking changes", "skip", "skip versioning", "skip v"}
+	commitChangeTypesMajorUpgrade = []string{"breaking change", "breaking changes"}
+	commitChangeTypesMinorUpgrade = []string{"feat"}
+	commitChangeTypePatchUpgrade  = []string{"build", "ci", "docs", "fix", "perf", "refactor", "style", "test"}
+	commitTypeSkipVersioning      = []string{"skip", "skip versioning", "skip v"}
+)
+
+type VersionControl interface {
```

Comment:
why do you have this interface where it is implemented... usually we define interface near where we need some abstraction... @esequielvirtuoso

Reply:
The interface has been removed.
As we discussed earlier, it was redundant with insufficient grounds.